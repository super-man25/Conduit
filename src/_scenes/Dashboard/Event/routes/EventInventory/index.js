// @flow

import styled from 'styled-components';
import React from 'react';
import {
  PageWrapper,
  Spacing,
  Box,
  Flex,
  FlexItem,
  PrimaryButton
} from '_components';
import { connect } from 'react-redux';
import EventHeader from '../../components/EventHeader';
import { selectors as eventSelectors } from '_state/event';
import { selectors as eventInventorySelectors } from '_state/eventInventory';
import {
  actions as eventInventoryBulkActions,
  selectors as eventInventoryBulkSelectors
} from '_state/eventInventoryBulk';
import { VirtualizedEventInventory } from './components/InventoryTable';
import type { EDEvent } from '_models';
import { EventInventorySeatMap } from './components/InventorySeatMap';
import { Portal } from '_components/Portal';
import { BulkUpdateModal } from './components/BulkUpdateModal';
import { createStructuredSelector } from 'reselect';

const EventInventoryTableContainer = styled(FlexItem)`
  min-height: 60%;
  min-width: 1100px;
`;

type Props = {
  event: ?EDEvent,
  selectedEventIds: number[],
  isBulkUpdating: boolean,
  startBulkUpdate: () => void
};

export const EventInventory = ({
  event,
  selectedEventIds,
  isBulkUpdating,
  startBulkUpdate
}: Props) =>
  !!event && (
    <PageWrapper>
      <Flex direction="column" height="100%">
        <Box padding="1.5rem 2rem">
          <EventHeader availableInventory={15000} totalInventory={40000} />
          <Box marginTop="1rem">
            <EventInventorySeatMap />
          </Box>
        </Box>
        <Spacing height="2rem" />
        <Box padding="0 1rem">
          <PrimaryButton
            disabled={!selectedEventIds.length}
            title="Update Selected Items"
            onClick={startBulkUpdate}
            small
          >
            Bulk Update ({selectedEventIds.length} Rows)
          </PrimaryButton>
        </Box>
        <EventInventoryTableContainer>
          <VirtualizedEventInventory event={event} />
        </EventInventoryTableContainer>
      </Flex>
      <Portal>
        {isBulkUpdating && (
          <BulkUpdateModal selectedEventIds={selectedEventIds} />
        )}
      </Portal>
    </PageWrapper>
  );

const mapStateToProps = createStructuredSelector({
  event: eventSelectors.selectEvent,
  selectedEventIds: eventInventorySelectors.selectSelectedRowIds,
  isBulkUpdating: eventInventoryBulkSelectors.isBulkUpdating
});

const mapDispatchToProps = {
  startBulkUpdate: eventInventoryBulkActions.startBulkUpdate,
  cancelBulkUpdate: eventInventoryBulkActions.cancelBulkUpdate
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(EventInventory);
