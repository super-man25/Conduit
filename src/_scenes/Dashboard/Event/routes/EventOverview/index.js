// @flow
import React from 'react';
import { PageWrapper, Spacing, Box } from '_components';
import { connect } from 'react-redux';
import EventHeader from '../../components/EventHeader';
import EventChart from './components/EventChart';
import type { EDEvent } from '_models';
import { selectors } from '_state/event';
import { TicketIntegrations } from './components/TicketIntegrations';
import { EventPricing } from './components/EventPricing';

type Props = {
  event: ?EDEvent,
  isAdmin: boolean
};

export const EventOverview = ({ event, isAdmin }: Props) =>
  !!event && (
    <PageWrapper>
      <Box padding="1.5rem 2rem">
        <EventHeader availableInventory={15000} totalInventory={40000} />
        <Spacing height="1rem" />
        <EventChart event={event} />
        <Spacing height="2rem" />
        <TicketIntegrations />
        <Spacing height="2rem" />
        {isAdmin && <EventPricing event={event} isAdmin={isAdmin} />}
      </Box>
    </PageWrapper>
  );

function mapStateToProps(state) {
  return {
    event: selectors.selectEvent(state),
    isAdmin: state.auth.model.isAdmin
  };
}

export default connect(mapStateToProps)(EventOverview);
