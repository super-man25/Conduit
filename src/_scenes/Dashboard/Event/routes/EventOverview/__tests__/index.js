import React from 'react';
import { shallow } from 'enzyme';
import { EventOverview } from '_scenes/Dashboard/Event/routes/EventOverview';

describe('<EventOverview /> index route', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<EventOverview event={{}} />);
    expect(wrapper).toMatchSnapshot();
  });
});
