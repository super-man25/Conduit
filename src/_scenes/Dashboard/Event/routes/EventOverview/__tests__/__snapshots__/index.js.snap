// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EventOverview /> index route should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <EventOverview
    event={Object {}}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <styled.div
        padding="1.5rem 2rem"
      >
        <Connect(withSidebarEventHeader)
          availableInventory={15000}
          totalInventory={40000}
        />
        <styled.div
          height="1rem"
        />
        <Connect(EventChart)
          event={Object {}}
        />
        <styled.div
          height="2rem"
        />
        <Connect(TicketIntegrationsPresenter) />
        <styled.div
          height="2rem"
        />
      </styled.div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Connect(withSidebarEventHeader)
            availableInventory={15000}
            totalInventory={40000}
          />,
          <styled.div
            height="1rem"
          />,
          <Connect(EventChart)
            event={Object {}}
          />,
          <styled.div
            height="2rem"
          />,
          <Connect(TicketIntegrationsPresenter) />,
          <styled.div
            height="2rem"
          />,
          undefined,
        ],
        "padding": "1.5rem 2rem",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "availableInventory": 15000,
            "totalInventory": 40000,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "height": "1rem",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "event": Object {},
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "height": "2rem",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "height": "2rem",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        undefined,
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <styled.div
          padding="1.5rem 2rem"
        >
          <Connect(withSidebarEventHeader)
            availableInventory={15000}
            totalInventory={40000}
          />
          <styled.div
            height="1rem"
          />
          <Connect(EventChart)
            event={Object {}}
          />
          <styled.div
            height="2rem"
          />
          <Connect(TicketIntegrationsPresenter) />
          <styled.div
            height="2rem"
          />
        </styled.div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Connect(withSidebarEventHeader)
              availableInventory={15000}
              totalInventory={40000}
            />,
            <styled.div
              height="1rem"
            />,
            <Connect(EventChart)
              event={Object {}}
            />,
            <styled.div
              height="2rem"
            />,
            <Connect(TicketIntegrationsPresenter) />,
            <styled.div
              height="2rem"
            />,
            undefined,
          ],
          "padding": "1.5rem 2rem",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "availableInventory": 15000,
              "totalInventory": 40000,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "height": "1rem",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "event": Object {},
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "height": "2rem",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "height": "2rem",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          undefined,
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
